applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_analysis.py

```diff
@@ -1,21 +1,8 @@
-# Importing Kratos
-import KratosMultiphysics
-
-# Importing the base class
 from KratosMultiphysics.analysis_stage import AnalysisStage
+from KratosMultiphysics.CompressiblePotentialFlowApplication.python_solvers_wrapper_compressible_potential import CreateSolverByParameters
 
 class PotentialFlowAnalysis(AnalysisStage):
     '''Main script for potential flow simulations.'''
 
     def _CreateSolver(self):
-        if self.project_parameters["solver_settings"]["solver_type"].GetString()=="potential_flow":
-            import KratosMultiphysics.CompressiblePotentialFlowApplication.potential_flow_solver as potential_flow_solver
-            return potential_flow_solver.CreateSolver(self.model, self.project_parameters["solver_settings"])
-        elif self.project_parameters["solver_settings"]["solver_type"].GetString()=="ale_potential_flow":
-            import KratosMultiphysics.CompressiblePotentialFlowApplication.ale_potential_flow_solver as ale_potential_flow_solver
-            return ale_potential_flow_solver.CreateSolver(self.model, self.project_parameters["solver_settings"], self.parallel_type)
-        elif self.project_parameters["solver_settings"]["solver_type"].GetString()=="adjoint_potential_flow":
-            import KratosMultiphysics.CompressiblePotentialFlowApplication.potential_flow_adjoint_solver as adjoint_solver
-            return adjoint_solver.CreateSolver(self.model, self.project_parameters["solver_settings"])
-        else:
-            raise Exception("Solver type '"+str(self.project_parameters["solver_settings"]["solver_type"].GetString())+"' not added. Please specify an available solver")
+        return CreateSolverByParameters(self.model, self.project_parameters["solver_settings"], self.parallel_type)
```

Comment:
I just checked the fluid analysis and there the function CreateSolver(model, params) is used instead, which then calls the CreateSolverByParams() function:

https://github.com/KratosMultiphysics/Kratos/blob/3badd462572ba10c8c8189d7f23df034a14762eb/applications/FluidDynamicsApplication/python_scripts/python_solvers_wrapper_fluid.py#L69-L80


Could you add this to the solver wrapper and call this function to keep the same structure as the fluid app?

Reply:
No problem